library/x509_create.c

```diff
@@ -116,16 +117,21 @@ int mbedtls_x509_string_to_names( mbedtls_asn1_named_data **head, const char *na
 
         if( !in_tag && *c == '\\' && c != end )
```

Comment:
!in_tag could be extracted as a separate condition above, and these three separate checks for it would be unnecessary.